test/core.jl

```diff
@@ -2238,7 +2238,7 @@ f9534d(x) = (a=(1,2,4,6,7); a[x])
 @test try; f9534d() catch ex; (ex::BoundsError).a === (1,2,4,6,7) && ex.i == 7; end
 @test try; f9534d(-1) catch ex; (ex::BoundsError).a === (1,2,4,6,7) && ex.i == -1; end
 f9534e(x) = (a=IOBuffer(); setfield!(a, x, 3))
-@test try; f9534e(-2) catch ex; isa((ex::BoundsError).a,Base.IOBuffer) && ex.i == -2; end
+@test try; f9534e(-2) catch ex; (ex::BoundsError).a == Base.IOBuffer && ex.i == -2; end
```

Comment:
I'm not sure if there is a clean way to fix this.

If we had an `isfield(T, i)` intrinsic, I could check it explicitly and then throw an error, but we don't have such an intrinsic.  I can replace `fieldtype(T, i)` with `try; fieldtype(T, i); catch; Any; end` — that works, but is mildly horrifying.

Suggestions?